Sources/RabbitMq/Consumer/Consumer.swift

```diff
@@ -0,0 +1,47 @@
+@preconcurrency import AMQPClient
+import AsyncAlgorithms
+import Logging
+import NIO
+
+public struct Consumer: Sendable {
+    let connection: Connection
+    let configuration: ConsumerConfiguration
+    let logger: Logger
+
+    public init(
+        _ connection: Connection,
+        _ queueName: String,
+        _ exchangeName: String = "",
+        _ routingKey: String = "",
+        exchangeOptions: ExchangeOptions = ExchangeOptions(),
+        queueOptions: QueueOptions = QueueOptions(),
+        bindingOptions: BindingOptions = BindingOptions(),
+        consumerOptions: ConsumerOptions = ConsumerOptions()
+    ) {
+        self.connection = connection
+        self.configuration = ConsumerConfiguration(
+            queueName: queueName,
+            exchangeName: exchangeName,
+            routingKey: routingKey,
+            exchangeOptions: exchangeOptions,
+            queueOptions: queueOptions,
+            bindingOptions: bindingOptions,
+            consumerOptions: consumerOptions
+        )
+        self.logger = connection.logger
+    }
+
+    public func consume() async throws -> AnyAsyncSequence<String> {
```

Comment:
I wonder if this could be returning `some AsyncSequence<String>` now with Swift 6.

Reply:
I would like to not do this here. I want the library to support Swift 5.9 and above, and not be tied to Swift 6. This isn't like our internal projects which only use Swift 6 and above.